ci/do_ci.sh

```diff
@@ -679,6 +679,35 @@ case $CI_TARGET in
         bazel_envoy_binary_build sizeopt
         ;;
 
+    trigger_verify)
+        setup_clang_toolchain
+        WORKFLOW="envoy-verify.yml"
+        # This may be a merge commit in a PR
+        COMMIT="$(git rev-parse HEAD)"
+        # For a PR this should be the last actual commit
+        ENVOY_COMMIT="${ENVOY_COMMIT:-${COMMIT}}"
+        ENVOY_REPO="${ENVOY_REPO:-envoyproxy/envoy}"
+        # ENVOY_HEAD_REF must also be set in PRs to provide a unique key for job grouping and cancellation.
+        echo "Trigger workflow (${WORKFLOW})"
+        echo "  Repo: ${ENVOY_REPO}"
+        echo "  Branch: ${ENVOY_BRANCH}"
+        echo "  Inputs:"
+        echo "    ref: ${COMMIT}"
+        echo "    sha: ${ENVOY_COMMIT}"
+        echo "    head_ref: ${ENVOY_HEAD_REF}"
```

Comment:
Would be better to add some more comments here to explain the diffference of `ref` and `sha`.

And for the post submit, the ENVOY_HEAD_REF will no be set?

```
        ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
          ENVOY_BRANCH: "$(System.PullRequest.TargetBranch)"
          ENVOY_COMMIT: "$(System.PullRequest.SourceCommitId)"
          ENVOY_HEAD_REF: "$(Build.SourceBranch)"
        ${{ if ne(variables['Build.Reason'], 'PullRequest') }}:
          ENVOY_BRANCH: "$(Build.SourceBranch)"
```

Reply:
ive added the explainer in `ci/do_ci.sh` and a stub here pointing to it